### ISSUE TITLE
Multi-target test run chooses the incorrect target

### ISSUE THREAD

**Issue by bridadan** [ARMmbed/greentea#187]
When doing a multi-target test run, greentea will incorrectly call `mbedhtrun` with the only one of the target's information. For example:

```
$ mbedgt -V -v -t K64F-GCC_ARM,NUCLEO_F401RE-GCC_ARM
mbedgt: greentea test automation tool ver. 1.2.0
mbedgt: using multiple test specifications from current directory!
        using '.build\tests\K64F\ARM\test_spec.json'
        using '.build\tests\K64F\GCC_ARM\test_spec.json'
        using '.build\tests\KL25Z\GCC_ARM\test_spec.json'
        using '.build\tests\KL46Z\GCC_ARM\test_spec.json'
        using '.build\tests\NRF51_DK\GCC_ARM\test_spec.json'
        using '.build\tests\NUCLEO_F401RE\GCC_ARM\test_spec.json'
mbedgt: detecting connected mbed-enabled devices...
mbedgt: detected 2 devices
        +---------------+----------------------+-------------+-------------+--------------------------------------------------+
        | platform_name | platform_name_unique | serial_port | mount_point | target_id                                        |
        +---------------+----------------------+-------------+-------------+--------------------------------------------------+
        | NUCLEO_F401RE | NUCLEO_F401RE[0]     | COM273      | D:          | 072002210774650F3830F312                         |
        | K64F          | K64F[0]              | COM255      | E:          | 0240000028884e450037700f6bf0000d8021000097969900 |
        +---------------+----------------------+-------------+-------------+--------------------------------------------------+
mbedgt: processing target 'K64F' toolchain 'GCC_ARM' compatible platforms... (note: switch set to --parallel 1)
        +---------------+----------------------+-------------+-------------+--------------------------------------------------+
        | platform_name | platform_name_unique | serial_port | mount_point | target_id                                        |
        +---------------+----------------------+-------------+-------------+--------------------------------------------------+
        | K64F          | K64F[0]              | COM255:9600 | E:          | 0240000028884e450037700f6bf0000d8021000097969900 |
        +---------------+----------------------+-------------+-------------+--------------------------------------------------+
mbedgt: running 2 tests for platform 'K64F' and toolchain 'GCC_ARM'
        use 1 instance of execution threads for testing
mbedgt: checking for 'host_tests' directory above image directory structure
        found 'host_tests' directory in: 'TESTS\host_tests'
mbedgt: selecting test case observer...
        calling mbedhtrun: mbedhtrun -m K64F -p COM255:9600 -f ".build/tests/K64F/GCC_ARM/TESTS/mbedmicro-rtos-mbed/basic/basic.bin" -d E: -C 4 -c shell -t 0240000028884e450037700f6bf0000d802100009796
9900 -e "TESTS\host_tests"
mbedgt: mbed-host-test-runner: started
[1473181713.63][HTST][INF] host test executor ver. 1.1.0
[1473181713.63][HTST][INF] copy image onto target...
[1473181713.63][COPY][INF] Waiting up to 60 sec for '0240000028884e450037700f6bf0000d8021000097969900' mount point (current is 'E:')...
        1 file(s) copied.
[1473181722.36][HTST][INF] starting host test process...
[1473181722.78][CONN][INF] starting connection process...
[1473181722.78][CONN][INF] notify event queue about extra 60 sec timeout for serial port pooling
[1473181722.78][CONN][INF] initializing serial port listener...
[1473181722.78][HTST][INF] setting timeout to: 60 sec
[1473181722.78][SERI][INF] serial(port=COM255, baudrate=9600, timeout=0, write_timeout=5)
[1473181722.78][PLGN][INF] Waiting up to 60 sec for '0240000028884e450037700f6bf0000d8021000097969900' serial port (current is 'COM255')...
[1473181723.01][SERI][INF] reset device using 'default' plugin...
[1473181723.26][SERI][INF] waiting 1.00 sec after reset
[1473181724.26][SERI][INF] wait for it...
[1473181724.26][SERI][TXD] mbedmbedmbedmbedmbedmbedmbedmbedmbedmbed
[1473181724.26][CONN][INF] sending up to 2 __sync packets (specified with --sync=2)
[1473181724.26][CONN][INF] sending preamble '99269959-cfd4-4eae-bdb1-206d4ac5e1c5'
[1473181724.26][SERI][TXD] {{__sync;99269959-cfd4-4eae-bdb1-206d4ac5e1c5}}
[1473181724.40][CONN][RXD] {{__sync;99269959-cfd4-4eae-bdb1-206d4ac5e1c5}}
[1473181724.40][CONN][INF] found SYNC in stream: {{__sync;99269959-cfd4-4eae-bdb1-206d4ac5e1c5}} it is #0 sent, queued...
[1473181724.40][HTST][INF] sync KV found, uuid=99269959-cfd4-4eae-bdb1-206d4ac5e1c5, timestamp=1473181724.402000
[1473181724.42][CONN][RXD] {{__version;1.3.0}}
[1473181724.42][CONN][INF] found KV pair in stream: {{__version;1.3.0}}, queued...
[1473181724.42][HTST][INF] DUT greentea-client version: 1.3.0
[1473181724.44][CONN][RXD] {{__timeout;15}}
[1473181724.44][CONN][INF] found KV pair in stream: {{__timeout;15}}, queued...
[1473181724.44][HTST][INF] setting timeout to: 15 sec
[1473181724.48][CONN][RXD] {{__host_test_name;wait_us_auto}}
[1473181724.48][CONN][INF] found KV pair in stream: {{__host_test_name;wait_us_auto}}, queued...
[1473181724.48][HTST][INF] host test class: '<class 'mbed_host_tests.host_tests.wait_us_auto.WaitusTest'>'
[1473181724.48][HTST][INF] host test setup() call...
[1473181724.48][HTST][INF] CALLBACKs updated
[1473181724.48][HTST][INF] host test detected: wait_us_auto
[1473181725.49][CONN][RXD] {{tick;0}}
[1473181725.49][CONN][INF] found KV pair in stream: {{tick;0}}, queued...
[1473181725.49][HTST][INF] tick! 1473181725.49
[1473181726.49][CONN][RXD] {{tick;1}}
[1473181726.49][CONN][INF] found KV pair in stream: {{tick;1}}, queued...
[1473181726.49][HTST][INF] tick! 1473181726.49
[1473181727.50][CONN][RXD] {{tick;2}}
[1473181727.50][CONN][INF] found KV pair in stream: {{tick;2}}, queued...
[1473181727.50][HTST][INF] tick! 1473181727.5
[1473181728.51][CONN][RXD] {{tick;3}}
[1473181728.51][CONN][INF] found KV pair in stream: {{tick;3}}, queued...
[1473181728.51][HTST][INF] tick! 1473181728.51
[1473181729.52][CONN][RXD] {{tick;4}}
[1473181729.52][CONN][INF] found KV pair in stream: {{tick;4}}, queued...
[1473181729.52][HTST][INF] tick! 1473181729.52
[1473181730.53][CONN][RXD] {{tick;5}}
[1473181730.53][CONN][INF] found KV pair in stream: {{tick;5}}, queued...
[1473181730.53][HTST][INF] tick! 1473181730.53
[1473181731.54][CONN][RXD] {{tick;6}}
[1473181731.54][CONN][INF] found KV pair in stream: {{tick;6}}, queued...
[1473181731.54][HTST][INF] tick! 1473181731.54
[1473181732.55][CONN][RXD] {{tick;7}}
[1473181732.55][CONN][INF] found KV pair in stream: {{tick;7}}, queued...
[1473181732.55][HTST][INF] tick! 1473181732.55
[1473181733.56][CONN][RXD] {{tick;8}}
[1473181733.56][CONN][INF] found KV pair in stream: {{tick;8}}, queued...
[1473181733.56][HTST][INF] tick! 1473181733.56
[1473181734.57][CONN][RXD] {{tick;9}}
[1473181734.57][CONN][INF] found KV pair in stream: {{tick;9}}, queued...
[1473181734.57][HTST][INF] tick! 1473181734.57
[1473181735.58][CONN][RXD] {{tick;10}}
[1473181735.58][CONN][INF] found KV pair in stream: {{tick;10}}, queued...
[1473181735.58][HTST][INF] tick! 1473181735.58
[1473181736.59][CONN][RXD] {{tick;11}}
[1473181736.59][CONN][INF] found KV pair in stream: {{tick;11}}, queued...
[1473181736.59][HTST][INF] tick! 1473181736.59
[1473181737.60][CONN][RXD] {{tick;12}}
[1473181737.60][CONN][INF] found KV pair in stream: {{tick;12}}, queued...
[1473181737.60][HTST][INF] tick! 1473181737.6
[1473181738.61][CONN][RXD] {{tick;13}}
[1473181738.61][CONN][INF] found KV pair in stream: {{tick;13}}, queued...
[1473181738.61][HTST][INF] tick! 1473181738.61
[1473181739.44][HTST][INF] test suite run finished after 15.00 sec...
[1473181739.44][CONN][INF] received special even '__host_test_finished' value='True', finishing
[1473181739.45][HTST][INF] CONN exited with code: 0
[1473181739.45][HTST][INF] No events in queue
[1473181739.45][HTST][INF] stopped consuming events
[1473181739.45][HTST][INF] host test result(): True
[1473181739.45][HTST][WRN] missing __exit event from DUT
[1473181739.45][HTST][WRN] missing __exit_event_queue event from host test
[1473181739.45][HTST][INF] calling blocking teardown()
[1473181739.45][HTST][INF] teardown() finished
[1473181739.45][HTST][INF] {{result;success}}
mbedgt: checking for GCOV data...
mbedgt: mbed-host-test-runner: stopped and returned 'OK'
mbedgt: test case summary event not found
        no test case report present, assuming test suite to be a single test case!
        test suite: tests-mbedmicro-rtos-mbed-basic
        test case: tests-mbedmicro-rtos-mbed-basic
mbedgt: test on hardware with target id: 0240000028884e450037700f6bf0000d8021000097969900
mbedgt: test suite 'tests-mbedmicro-rtos-mbed-basic' ................................................. OK in 26.20 sec
        test case: 'tests-mbedmicro-rtos-mbed-basic' ................................................. OK in 26.20 sec
mbedgt: test case summary: 1 pass, 0 failures
mbedgt: checking for 'host_tests' directory above image directory structure
        found 'host_tests' directory in: 'features\net\FEATURE_IPV4\TESTS\mbedmicro-net\host_tests'
mbedgt: selecting test case observer...
        calling mbedhtrun: mbedhtrun -m K64F -p COM255:9600 -f ".build/tests/K64F/GCC_ARM/features/net/FEATURE_IPV4/TESTS/mbedmicro-net/udp_echo_client/udp_echo_client.bin" -d E: -C 4 -c shell -t 0240
000028884e450037700f6bf0000d8021000097969900 -e "features\net\FEATURE_IPV4\TESTS\mbedmicro-net\host_tests"
mbedgt: mbed-host-test-runner: started
[1473181739.87][HTST][INF] host test executor ver. 1.1.0
[1473181739.87][HTST][INF] copy image onto target...
[1473181739.87][COPY][INF] Waiting up to 60 sec for '0240000028884e450037700f6bf0000d8021000097969900' mount point (current is 'E:')...
        1 file(s) copied.
[1473181752.41][HTST][INF] starting host test process...
[1473181752.81][CONN][INF] starting connection process...
[1473181752.81][CONN][INF] notify event queue about extra 60 sec timeout for serial port pooling
[1473181752.82][CONN][INF] initializing serial port listener...
[1473181752.82][HTST][INF] setting timeout to: 60 sec
[1473181752.82][SERI][INF] serial(port=COM255, baudrate=9600, timeout=0, write_timeout=5)
[1473181752.82][PLGN][INF] Waiting up to 60 sec for '0240000028884e450037700f6bf0000d8021000097969900' serial port (current is 'COM255')...
[1473181753.04][SERI][INF] reset device using 'default' plugin...
[1473181753.29][SERI][INF] waiting 1.00 sec after reset
[1473181754.29][SERI][INF] wait for it...
[1473181754.29][SERI][TXD] mbedmbedmbedmbedmbedmbedmbedmbedmbedmbed
[1473181754.29][CONN][INF] sending up to 2 __sync packets (specified with --sync=2)
[1473181754.29][CONN][INF] sending preamble '1d8a3825-70dc-4acc-89b9-98a917eccd72'
[1473181754.29][SERI][TXD] {{__sync;1d8a3825-70dc-4acc-89b9-98a917eccd72}}
[1473181754.43][CONN][RXD] {{__sync;1d8a3825-70dc-4acc-89b9-98a917eccd72}}
[1473181754.44][CONN][INF] found SYNC in stream: {{__sync;1d8a3825-70dc-4acc-89b9-98a917eccd72}} it is #0 sent, queued...
[1473181754.44][HTST][INF] sync KV found, uuid=1d8a3825-70dc-4acc-89b9-98a917eccd72, timestamp=1473181754.436000
[1473181754.46][CONN][RXD] {{__version;1.3.0}}
[1473181754.46][CONN][INF] found KV pair in stream: {{__version;1.3.0}}, queued...
[1473181754.46][HTST][INF] DUT greentea-client version: 1.3.0
[1473181754.47][CONN][RXD] {{__timeout;20}}
[1473181754.47][CONN][INF] found KV pair in stream: {{__timeout;20}}, queued...
[1473181754.47][HTST][INF] setting timeout to: 20 sec
[1473181754.51][CONN][RXD] {{__host_test_name;udp_echo_client}}
[1473181754.51][CONN][INF] found KV pair in stream: {{__host_test_name;udp_echo_client}}, queued...
[1473181754.51][HTST][INF] host test class: '<class 'udp_echo_client.UDPEchoClientTest'>'
[1473181754.51][HTST][INF] host test setup() call...
[1473181754.51][HTST][INF] CALLBACKs updated
[1473181754.51][HTST][INF] host test detected: udp_echo_client
[1473181763.13][CONN][RXD] UDP client IP Address is 192.168.1.71
[1473181763.16][CONN][RXD] {{target_ip;192.168.1.71}}
[1473181763.16][CONN][INF] found KV pair in stream: {{target_ip;192.168.1.71}}, queued...
[1473181763.16][HTST][INF] HOST: Listening for UDP packets: 10.118.14.45:51444
[1473181763.17][CONN][RXD] {{host_ip; }}
[1473181763.17][CONN][INF] found KV pair in stream: {{host_ip; }}, queued...
[1473181763.17][SERI][TXD] {{host_ip;10.118.14.45}}
[1473181763.22][CONN][RXD] {{host_port; }}
[1473181763.22][CONN][INF] found KV pair in stream: {{host_port; }}, queued...
[1473181763.22][SERI][TXD] {{host_port;51444}}
[1473181763.30][CONN][RXD] MBED: UDP Server IP address received: 10.118.14.45:51444
[1473181763.32][CONN][RXD] [00] sent...256 Bytes
[1473181763.35][CONN][RXD] [00] recv...256 Bytes
[1473181763.37][CONN][RXD] [01] sent...256 Bytes
HOST: UDPEchoClientHandler: Rx:
32908266911358306927728039249170450402431066197510445907634409062230995597974696435769425381007563172682100811474758187173950032437100786378388538016709783304532072050062507418690552416285083969052345
03505554833401643079599077573652253337576769482150641004

HOST: UDPEchoClientHandler: Rx:
69749253262377307669817603840780440958119127245135862782743597627530861777809006139380439812144359578183959984311647251222314472431356966938747685449033472769284382478610288885993696602045119606335815
97767724679016230062645775830244125019710150628868684332

[1473181774.47][HTST][INF] test suite run finished after 20.00 sec...
[1473181774.47][CONN][INF] received special even '__host_test_finished' value='True', finishing
[1473181774.48][HTST][INF] CONN exited with code: 0
[1473181774.48][HTST][INF] No events in queue
[1473181774.48][HTST][INF] stopped consuming events
[1473181774.48][HTST][INF] host test result(): None
[1473181774.48][HTST][WRN] missing __exit event from DUT
[1473181774.48][HTST][WRN] missing __exit_event_queue event from host test
[1473181774.49][HTST][ERR] missing __exit_event_queue event from host test and no result from host test, timeout...
[1473181774.49][HTST][INF] calling blocking teardown()
[1473181774.85][HTST][INF] teardown() finished
[1473181774.85][HTST][INF] {{result;timeout}}
mbedgt: checking for GCOV data...
mbedgt: mbed-host-test-runner: stopped and returned 'TIMEOUT'
mbedgt: test case summary event not found
        no test case report present, assuming test suite to be a single test case!
        test suite: features-net-feature_ipv4-tests-mbedmicro-net-udp_echo_client
        test case: features-net-feature_ipv4-tests-mbedmicro-net-udp_echo_client
mbedgt: test on hardware with target id: 0240000028884e450037700f6bf0000d8021000097969900
mbedgt: test suite 'features-net-feature_ipv4-tests-mbedmicro-net-udp_echo_client' ................... TIMEOUT in 35.38 sec
        test case: 'features-net-feature_ipv4-tests-mbedmicro-net-udp_echo_client' ................... ERROR in 35.38 sec
mbedgt: test case summary: 0 passes, 1 failure
mbedgt: all tests finished!
mbedgt: processing target 'NUCLEO_F401RE' toolchain 'GCC_ARM' compatible platforms... (note: switch set to --parallel 1)
        +---------------+----------------------+-------------+-------------+--------------------------------------------------+
        | platform_name | platform_name_unique | serial_port | mount_point | target_id                                        |
        +---------------+----------------------+-------------+-------------+--------------------------------------------------+
        | K64F          | K64F[0]              | COM255:9600 | E:          | 0240000028884e450037700f6bf0000d8021000097969900 |
        | NUCLEO_F401RE | NUCLEO_F401RE[0]     | COM273:9600 | D:          | 072002210774650F3830F312                         |
        +---------------+----------------------+-------------+-------------+--------------------------------------------------+
mbedgt: running 1 test for platform 'NUCLEO_F401RE' and toolchain 'GCC_ARM'
        use 1 instance of execution threads for testing
mbedgt: checking for 'host_tests' directory above image directory structure
        found 'host_tests' directory in: 'TESTS\host_tests'
mbedgt: selecting test case observer...
        calling mbedhtrun: mbedhtrun -m K64F -p COM255:9600 -f ".build/tests/NUCLEO_F401RE/GCC_ARM/TESTS/mbedmicro-rtos-mbed/basic/basic.bin" -d E: -C 4 -c shell -t 0240000028884e450037700f6bf0000d802
1000097969900 -e "TESTS\host_tests"
mbedgt: mbed-host-test-runner: started
[1473181775.24][HTST][INF] host test executor ver. 1.1.0
[1473181775.24][HTST][INF] copy image onto target...
[1473181775.24][COPY][INF] Waiting up to 60 sec for '0240000028884e450037700f6bf0000d8021000097969900' mount point (current is 'E:')...
        1 file(s) copied.
[1473181779.64][HTST][INF] starting host test process...
[1473181780.02][CONN][INF] starting connection process...
[1473181780.02][CONN][INF] notify event queue about extra 60 sec timeout for serial port pooling
[1473181780.02][CONN][INF] initializing serial port listener...
[1473181780.03][SERI][INF] serial(port=COM255, baudrate=9600, timeout=0, write_timeout=5)
[1473181780.02][HTST][INF] setting timeout to: 60 sec
[1473181780.03][PLGN][INF] Waiting up to 60 sec for '0240000028884e450037700f6bf0000d8021000097969900' serial port (current is 'COM255')...
[1473181780.25][SERI][INF] reset device using 'default' plugin...
[1473181780.50][SERI][INF] waiting 1.00 sec after reset
[1473181781.50][SERI][INF] wait for it...
[1473181781.50][SERI][TXD] mbedmbedmbedmbedmbedmbedmbedmbedmbedmbed
[1473181781.51][CONN][INF] sending up to 2 __sync packets (specified with --sync=2)
[1473181781.51][CONN][INF] sending preamble '915e442d-6423-44d1-b433-ddf57f3f0e0e'
[1473181781.51][SERI][TXD] {{__sync;915e442d-6423-44d1-b433-ddf57f3f0e0e}}
[1473181781.65][CONN][RXD] {{__sync;915e442d-6423-44d1-b433-ddf57f3f0e0e}}
[1473181781.65][CONN][INF] found SYNC in stream: {{__sync;915e442d-6423-44d1-b433-ddf57f3f0e0e}} it is #0 sent, queued...
[1473181781.65][HTST][INF] sync KV found, uuid=915e442d-6423-44d1-b433-ddf57f3f0e0e, timestamp=1473181781.647000
[1473181781.67][CONN][RXD] {{__version;1.3.0}}
[1473181781.67][CONN][INF] found KV pair in stream: {{__version;1.3.0}}, queued...
[1473181781.67][HTST][INF] DUT greentea-client version: 1.3.0
[1473181781.69][CONN][RXD] {{__timeout;20}}
[1473181781.69][CONN][INF] found KV pair in stream: {{__timeout;20}}, queued...
[1473181781.69][HTST][INF] setting timeout to: 20 sec
[1473181781.72][CONN][RXD] {{__host_test_name;udp_echo_client}}
[1473181781.72][CONN][INF] found KV pair in stream: {{__host_test_name;udp_echo_client}}, queued...
[1473181781.72][HTST][ERR] host test not detected: udp_echo_client
[1473181781.72][HTST][INF] __exit_event_queue received
[1473181781.72][HTST][INF] test suite run finished after 0.04 sec...
[1473181781.72][CONN][INF] received special even '__host_test_finished' value='True', finishing
[1473181781.74][HTST][INF] CONN exited with code: 0
[1473181781.74][HTST][INF] No events in queue
[1473181781.74][HTST][INF] stopped consuming events
[1473181781.74][HTST][INF] host test result() call skipped, received: error
[1473181781.74][HTST][WRN] missing __exit event from DUT
[1473181781.74][HTST][INF] calling blocking teardown()
[1473181781.74][HTST][INF] teardown() finished
[1473181781.74][HTST][INF] {{result;error}}
mbedgt: checking for GCOV data...
mbedgt: mbed-host-test-runner: stopped and returned 'ERROR'
mbedgt: test case summary event not found
        no test case report present, assuming test suite to be a single test case!
        test suite: tests-mbedmicro-rtos-mbed-basic
        test case: tests-mbedmicro-rtos-mbed-basic
mbedgt: test on hardware with target id: 0240000028884e450037700f6bf0000d8021000097969900
mbedgt: test suite 'tests-mbedmicro-rtos-mbed-basic' ................................................. ERROR in 6.88 sec
        test case: 'tests-mbedmicro-rtos-mbed-basic' ................................................. ERROR in 6.88 sec
mbedgt: test case summary: 0 passes, 1 failure
mbedgt: all tests finished!
mbedgt: shuffle seed: 0.3178166174
mbedgt: exporting to HTML file 'report.html'...
mbedgt: unexpected error:
        u'features-net-feature_ipv4-tests-mbedmicro-net-udp_echo_client'
```

You can see that `mbedhtrun` is copying the correct binary over each time, but it is copying it to the incorrect target (eg a NUCLEO_F401RE binary onto a K64F).

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSFW-2898

**Comment by bridadan** [comment]
Should be fixed with the merge of #189
